### PR TITLE
core: split system/user job timeouts and make them configurable

### PR DESCRIPTION
**Pull request by keszybz**
Config options are -Ddefault-timeout-sec= and -Ddefault-user-timeout-sec=. Existing -Dupdate-helper-user-timeout= is renamed to -Dupdate-helper-user-timeout-sec= for consistency. All three options take an integer value in seconds. The renaming and type-change of the option is a small compat break, but it's just at compile time and result in a clear error message. I also doubt that anyone was actually using the option.

This commit separates the user manager timeouts, but keeps them unchanged at 90 s. The timeout for the user manager is set to 4/3*user-timeout, which means that it is still 120 s. Similarly, the restart timeout is set to 10/9*timeout.

Fedora wants to experiment with lower timeouts, but doing this via a patch would be annoying and more work than necessary. Let's make this easy to configure.

### CODE REVIEW

File: src/core/main.c [hunk 1]
```diff
@@ -2436,9 +2436,9 @@ static void reset_arguments(void) {
...
         arg_default_std_error = EXEC_OUTPUT_INHERIT;
         arg_default_restart_usec = DEFAULT_RESTART_USEC;
-        arg_default_timeout_start_usec = DEFAULT_TIMEOUT_USEC;
-        arg_default_timeout_stop_usec = DEFAULT_TIMEOUT_USEC;
-        arg_default_timeout_abort_usec = DEFAULT_TIMEOUT_USEC;
+        arg_default_timeout_start_usec = manager_default_timeout(arg_system);
+        arg_default_timeout_stop_usec = manager_default_timeout(arg_system);
+        arg_default_timeout_abort_usec = manager_default_timeout(arg_system);
         arg_default_timeout_abort_set = false;
         arg_default_device_timeout_usec = DEFAULT_TIMEOUT_USEC;
```
Comment (poettering) [inline src/core/main.c]: why not give this the same treatment?

File: src/core/manager.c [hunk 2]
```diff
@@ -835,8 +835,8 @@ int manager_new(LookupScope scope, ManagerTestRunFlags test_run_flags, Manager *
                 .default_memory_accounting = MEMORY_ACCOUNTING_DEFAULT,
                 .default_tasks_accounting = true,
                 .default_tasks_max = TASKS_MAX_UNSET,
-                .default_timeout_start_usec = DEFAULT_TIMEOUT_USEC,
-                .default_timeout_stop_usec = DEFAULT_TIMEOUT_USEC,
+                .default_timeout_start_usec = manager_default_timeout(scope == LOOKUP_SCOPE_SYSTEM),
+                .default_timeout_stop_usec = manager_default_timeout(scope == LOOKUP_SCOPE_SYSTEM),
                 .default_restart_usec = DEFAULT_RESTART_USEC,
                 .default_device_timeout_usec = DEFAULT_TIMEOUT_USEC,
```
Comment (poettering) [inline src/core/manager.c]: same here

File: src/core/user.conf.in [hunk 3]
```diff
@@ -24,8 +24,8 @@
 #DefaultTimerAccuracySec=1min
 #DefaultStandardOutput=inherit
 #DefaultStandardError=inherit
-#DefaultTimeoutStartSec=90s
-#DefaultTimeoutStopSec=90s
+#DefaultTimeoutStartSec={{DEFAULT_USER_TIMEOUT_SEC}}s
+#DefaultTimeoutStopSec={{DEFAULT_USER_TIMEOUT_SEC}}s
```
Comment (poettering) [inline src/core/user.conf.in:28]: hmm, that DefaultDeviceTieoutSec= is not listed here seems like an accident, can you add it too?

File: meson_options.txt [hunk 4]
```diff
@@ -187,6 +187,10 @@ option('dbussystemservicedir', type : 'string',
        description : 'D-Bus system service directory')
 option('dbus-interfaces-dir', type : 'string',
        description : 'export D-Bus introspection XML as standalone files')
+option('default-timeout-sec', type : 'integer', value : 90,
+       description : 'default timeout for system unit start/stop')
+option('default-user-timeout-sec', type : 'integer', value : 90,
```
Comment (poettering) [inline meson_options.txt:192]: maybe cease the chance and lower the user service timeout to 30s at the same time?

**Comment by poettering** [comment]
i'd probably be more aggressive here, and also lower the user service timeout to 30s and announce it in NEWS, and see what happens. given this is now compile-time configurable, conservative distros such as debian can revert to 90s for that too

**Comment by bluca** [comment]
> i'd probably be more aggressive here, and also lower the user service timeout to 30s and announce it in NEWS, and see what happens. given this is now compile-time configurable, conservative distros such as debian can revert to 90s for that too

Let's postpone that to 254 maybe, given we are past rc1?

**Comment by keszybz** [comment]
https://github.com/systemd/systemd/pull/18386 tries to reduce the timeouts. Let's continue the discussion about specific values there.

This PR doesn't touch the timeouts on purpose, just make it easier to set them. I want it to go in for v253, and then we can experiments with the timeouts for v254 (also taking into account the feedback from Fedora).

Updated with the requested change to also cover DefaultDeviceTimeout=.

**Comment by keszybz** [comment]
jammy-amd64: times out in testsuite-70.sh.
jammy-ppc64el: subprocess.CalledProcessError: Command '['resolvectl', 'query', 'math.lab']' returned non-zero exit status 1. in `networkd-test.py`.
jammy-s390x: ERROR (CommandError): Unable to delete the specified server(s). testbed failure.

All three seem unrelated.

**Review by bluca — APPROVED** [review]

**Pull request merged**